### ISSUE TITLE
Blank line after node anchor: YAMLSemanticError: Nested mappings are not allowed in compact mappings

### ISSUE THREAD

**Issue by jdufresne** [eemeli/yaml#242]
**Describe the bug**

When there is a a blank line after a node anchor, `YAMLSemanticError` is thrown. If I understand the YAML spec correctly, nothing prohibits this.

**To Reproduce**

```js
import YAML from 'yaml'

let result = YAML.parse('key1: &default\n\n  subkey1: value1\n\nkey2:\n  <<: *default\n')
console.log(result)
```

**Expected behaviour**

```
{ key1: { subkey1: 'value1' }, key2: { subkey1: 'value1' } }
```

**Actual behaviour**

```
$ node index.mjs 
/home/jon/test/node_modules/yaml/dist/resolve/resolveMap.js:144
            doc.errors.push(new errors.YAMLSemanticError(item.node, msg));
                            ^

YAMLSemanticError: Nested mappings are not allowed in compact mappings at line 1, column 7:

key1: &default
      ^^^^^^^^…

    at resolveBlockMapItems (/home/jon/test/node_modules/yaml/dist/resolve/resolveMap.js:144:29)
    at Object.resolveMap (/home/jon/test/node_modules/yaml/dist/resolve/resolveMap.js:17:78)
    at Object.resolveTag (/home/jon/test/node_modules/yaml/dist/resolve/resolveTag.js:57:28)
    at resolveNodeValue (/home/jon/test/node_modules/yaml/dist/resolve/resolveNode.js:102:34)
    at Object.resolveNode (/home/jon/test/node_modules/yaml/dist/resolve/resolveNode.js:156:15)
    at Object.parseContents (/home/jon/test/node_modules/yaml/dist/doc/parseContents.js:24:31)
    at Document.parse (/home/jon/test/node_modules/yaml/dist/doc/Document.js:221:19)
    at new Document (/home/jon/test/node_modules/yaml/dist/doc/Document.js:51:12)
    at parseDocument (/home/jon/test/node_modules/yaml/dist/index.js:26:15)
    at Object.parse (/home/jon/test/node_modules/yaml/dist/index.js:41:15) {
  nodeType: 'MAP',
  range: Range { start: 6, end: 34 },
  linePos: { start: { line: 1, col: 7 }, end: { line: 4, col: 1 } }
}
```

**Versions (please complete the following information):**

- Environment: Node.js 14.6.0
- `yaml`: 2.0.0-3

**Additional context**

The blank lines are added for readability and as a team style choice.

If one adds a comment above the subkey, a different error is thrown: "YAMLSyntaxError: All collection items must start at the same"

```js
import YAML from 'yaml'

let result = YAML.parse('key1: &default\n\n  # This key ...\n  subkey1: value1\n\nkey2:\n  <<: *default\n')
console.log(result)
``` 

```
$ node index.mjs 
/home/jon/test/node_modules/yaml/dist/cst/Collection.js:170
          this.error = new errors.YAMLSyntaxError(this, msg);
                       ^

YAMLSyntaxError: All collection items must start at the same column at line 1, column 1:

key1: &default
^^^^^^^^^^^^^^…

    at Collection.parse (/home/jon/test/node_modules/yaml/dist/cst/Collection.js:170:24)
    at /home/jon/test/node_modules/yaml/dist/cst/ParseContext.js:137:29
    at Document.parseContents (/home/jon/test/node_modules/yaml/dist/cst/Document.js:164:26)
    at Document.parse (/home/jon/test/node_modules/yaml/dist/cst/Document.js:227:19)
    at Object.parse (/home/jon/test/node_modules/yaml/dist/cst/parse.js:24:18)
    at parseDocument (/home/jon/test/node_modules/yaml/dist/index.js:25:23)
    at Object.parse (/home/jon/test/node_modules/yaml/dist/index.js:41:15)
    at file:///home/jon/test/index.mjs:3:19
    at ModuleJob.run (internal/modules/esm/module_job.js:152:23)
    at async Loader.import (internal/modules/esm/loader.js:166:24) {
  nodeType: 'MAP',
  range: Range { start: 0, end: 73 },
  linePos: { start: { line: 1, col: 1 }, end: { line: 8, col: 1 } }
}
```

**Comment by eemeli** [comment]
That's really rather odd. It's fixed in the current `master` via the new parser, but that doesn't really help for Prettier. Commented about this more in the linked issue.

**Comment by eemeli** [comment]
Fixed in [`yaml@1.10.1`](https://github.com/eemeli/yaml/releases/tag/v1.10.1).
